### PR TITLE
do not construct std::string if there is no error

### PR DESCRIPTION
**Pull request by serxa**
Changelog category (leave one):
- Not for changelog (changelog entry is not required)

### CODE REVIEW

File: src/Core/Block.cpp [hunk 1]
```diff
@@ -38,14 +38,14 @@ static ReturnType onError(const std::string & message [[maybe_unused]], int code
 
 template <typename ReturnType>
 static ReturnType checkColumnStructure(const ColumnWithTypeAndName & actual, const ColumnWithTypeAndName & expected,
-    const std::string & context_description, bool allow_materialize, int code)
+    const char * context_description, bool allow_materialize, int code)
```
Comment (kitaisreal) [inline src/Core/Block.cpp:41]: It seems that std::string_view can be used here, without raw pointer.

**Comment by kitaisreal** [comment]
@Mergifyio update

**Review by alexey-milovidov — APPROVED** [review]

**Comment by kitaisreal** [comment]
@Mergifyio update

**Comment by kitaisreal** [comment]
@Mergifyio update

**Review by kitaisreal — APPROVED** [review]

**Pull request merged**